WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/StatsPage.java

```diff
@@ -15,28 +18,46 @@
 import static androidx.test.espresso.matcher.ViewMatchers.hasSibling;
 import static androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.isDescendantOfA;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
 import static org.hamcrest.Matchers.allOf;
 import static androidx.test.espresso.matcher.ViewMatchers.withText;
 import static org.wordpress.android.support.WPSupportUtils.dialogExistsWithTitle;
 import static org.wordpress.android.support.WPSupportUtils.scrollIntoView;
 import static org.wordpress.android.support.WPSupportUtils.tapButtonInDialogWithTitle;
-import static org.wordpress.android.support.WPSupportUtils.waitForElementToBeDisplayedWithoutFailure;
+import static org.wordpress.android.support.WPSupportUtils.waitForElementToBeDisplayed;
 
 public class StatsPage {
+    // We are interested only in "Stats" screen elements that are descendants of
+    // `coordinator_layout` which is actually shown on screen and contains user data.
+    // For whatever reason, there's also an instance of "Stats" screen template,
+    // which contains the cards headers, but has no user data loaded.
+    // It is located at the "right" of the one we need to work with:
+    //
+    // |--------------------|  |------------------------|
+    // | coordinator_layout |  |   coordinator_layout   |
+    // |Visible Stats Screen|  | Invisible Stats Screen |
+    // |--------------------|  |------------------------|
+    //
```

Comment:
I decided to add the explanation for this, because it's quite hard to guess otherwise, especially when using API < 29, which will not show the extra screens in the Layout Inspector (but they will still exist).

Reply:
Great, thanks for adding this bit of extra explanation here @pachlava ! 🥇